### ISSUE TITLE
cfn-lint incorrectly reports "E3682 Additional properties are not allowed" when using --configure-rule

### ISSUE THREAD

**Issue by alex-k86** [aws-cloudformation/cfn-lint#4178]
### CloudFormation Lint Version

cfn-lint 1.37.0

### What operating system are you using?

macOS 15.5

### Describe the bug

When invoking cfn-lint with the `--configure-rule "E3031:exceptions=^\${opt.*"` flag to provide an exception for the E3031 rule, an unrelated error `E3682 Additional properties are not allowed` is reported. This error is not present when the same CloudFormation template is linted without the `--configure-rule` flag. 

### Expected behavior

The error E3682 should not be reported when `--configure-rule` flag passed as a cli parameter 

### Reproduction template

```yaml
Parameters:
  MyDbPassword:
    Type: String
    NoEcho: true
Resources:
  PostgresInstance:
    Type: AWS::RDS::DBInstance
    DeletionPolicy: Delete
    Properties:
      Engine: postgres
      MasterUserPassword: !Ref MyDbPassword
```
